nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -160,147 +35,121 @@ impl DataStore {
+                is disabled",
+            );
+            bail!("Schema is out of date but automatic update is disabled");
+        };
+
         // If we're here, we know the following:
         //
         // - The schema does not match our expected version (or at least, it
-        // didn't when we read it moments ago).
+        //   didn't when we read it moments ago).
         // - We should attempt to automatically upgrade the schema.
-        //
-        // We do the following:
-        // - Look in the schema directory for all the changes, in-order, to
-        // migrate from our current version to the desired version.
+        info!(log, "Database schema is out of date.  Attempting upgrade.");
+        ensure!(
+            all_versions.contains_version(&found_version),
+            "Found schema version {found_version} was not found",
+        );
 
-        info!(log, "Reading schemas from {}", config.schema_dir);
-        let mut dir = tokio::fs::read_dir(&config.schema_dir)
-            .await
-            .map_err(|e| format!("Failed to read schema config dir: {e}"))?;
-        let mut all_versions = BTreeSet::new();
-        while let Some(entry) = dir
-            .next_entry()
-            .await
-            .map_err(|e| format!("Failed to read schema dir: {e}"))?
-        {
-            if entry.file_type().await.map_err(|e| e.to_string())?.is_dir() {
-                let name = entry
-                    .file_name()
-                    .into_string()
-                    .map_err(|_| "Non-unicode schema dir".to_string())?;
-                if let Ok(observed_version) = name.parse::<SemverVersion>() {
-                    all_versions.insert(observed_version);
-                } else {
-                    let err_msg =
-                        format!("Failed to parse {name} as a semver version");
-                    warn!(log, "{err_msg}");
-                    return Err(err_msg);
-                }
-            }
-        }
-
-        if !all_versions.contains(&current_version) {
-            return Err(format!(
-                "Current DB version {current_version} was not found in {}",
-                config.schema_dir
-            ));
-        }
         // TODO: Test this?
-        if !all_versions.contains(&desired_version) {
-            return Err(format!(
-                "Target DB version {desired_version} was not found in {}",
-                config.schema_dir
-            ));
-        }
-
-        let target_versions = all_versions.range((
-            Bound::Excluded(&current_version),
-            Bound::Included(&desired_version),
-        ));
-
+        ensure!(
+            all_versions.contains_version(&desired_version),
+            "Desired version {desired_version} was not found",
+        );
+
+        let target_versions: Vec<&SchemaVersion> = all_versions
+            .versions_range((
+                Bound::Excluded(&found_version),
+                Bound::Included(&desired_version),
```

Comment:
Tiny nitpick and I realize this didn't change, but do you think this would be clearer as `.versions_range(&found_version..=&desired_version)`? I don't usually see explicit `Bound` tuples.